plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/PropertyAwarePart.java

```diff
@@ -189,7 +189,9 @@ private void handleInputChange(PropertyChangeEvent evt) {
             //add new connection
             ConnectionEditPart editPart = createOrFindConnection(newValue);
             int modelIndex = getModelTargetConnections().indexOf(newValue);
-            addTargetConnection(editPart, modelIndex);
+            if (modelIndex != -1) {
```

Comment:
```suggestion
            if (modelIndex >= 0) {
```

It just seems better